app/pathogen/arbovirus/analyze/recharts/change-in-median-seroprevalence-over-time-graph.tsx

```diff
@@ -124,9 +149,14 @@ export const ChangeInMedianSeroprevalenceOverTimeGraph = (): React.ReactNode =>
                 intervalEndDate: parseISO(dataPoint.sampleEndDate),
               },
             })),
-            groupingBucketSize: {
-              years: 5,
-            },
+            desiredBucketCount: 10,
+            validBucketSizes: [
+              { years: 5 },
+              { years: 4 },
+              { years: 3 },
+              { years: 2 },
+              { years: 1 }
+            ]
```

Comment:
Each bucket size has a certain cost because it has to go through the entire array of studies for each bucket size so I'm not using anything above 5 because I found that in practice anything above 5 is pretty rare to get picked.

Reading this again, I think this algorithm has room a lot of room for optimization so I'm going to come back later and see what can be done. It's `O(number of studies * number of total years spanned for studies * number of possible bucket sizes)` right now for those who are curious but I think I can do a lot better.